### PR TITLE
support DOI, publishing, deleting

### PR DESCRIPTION
**Pull request by martinobersteiner**
new features:
  - `Delete`-Button on deposit-page now works
  - publishing now works
  - new DOI-field to generate a DOI for the upload
  - includes some minor updates to upload-page as per last LLT-meeting
  - includes `invenio-app-rdm` v11 compatibilty

fixes:
  - update LOMToDatacite44Schema (this was missed when updating metadata-schema to conform with LOM-UIBK)
  - service.create doesn't use `draft_item.to_dict()` anymore due to its many side effects
  - fxed a couple of javascript-warnings

notes:
  - invenio's DOI-field also allows to use a previously existing DOI, not sure whether we have that use-case but I left it in for now
  - some errors are shown twice as of right now, this is better fixed by the `deeper-error-levels`-PR on `react-invenio-deposit` (once I get to finish that...)

### CODE REVIEW

**Comment by utnapischtim** [comment]
please add the PR description to the commit message

File: invenio_records_lom/resources/serializers/schemas.py [hunk 1]
```diff
@@ -342,11 +348,12 @@ def get_contributors(self, obj: LOMRecord):
         contributes = obj["metadata"].get("lifeCycle", {}).get("contribute", [])
         contributordicts = []
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/schemas.py:349]: minor: normally the type of the items is not placed in the name. contributors is enough.

File: invenio_records_lom/resources/serializers/schemas.py [hunk 2]
```diff
@@ -303,16 +303,16 @@ def get_identifiers(self, obj: LOMRecord):
...
-            entities.append(contribute.get("entity"))
+            entities.extend(contribute.get("entity", []))
         return [{"name": entity} for entity in entities]
 
     def get_titles(self, obj: LOMRecord):
         """Get list of title-dicts."""
-        title = obj["metadata"].get("general", {}).get("title", "")
-        if not title:
+        title = obj["metadata"].get("general", {}).get("title", None)
+        if title is None:
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/schemas.py:315]: question: why the change from `if not title` to `if title is None`?

File: invenio_records_lom/resources/resources.py [hunk 3]
```diff
@@ -8,10 +8,10 @@
 """LOM resources."""
 
 from flask_resources import route
-from invenio_drafts_resources.resources import RecordResource
+from invenio_rdm_records.resources import RDMRecordResource
 
 
-class LOMRecordResource(RecordResource):
+class LOMRecordResource(RDMRecordResource):
```
Comment (utnapischtim) [inline invenio_records_lom/resources/resources.py:14]: question: why was that change necessary?

File: invenio_records_lom/services/schemas/metadata.py [hunk 4]
```diff
@@ -18,50 +18,63 @@ class Meta:
 
         unknown = INCLUDE
 
+    def dump(self, obj, **_):
+        """Overwrite dump to return `obj`, bypassing validation."""
```
[the remark this answers is not in the document]
Comment (martinobersteiner) [reply invenio_records_lom/services/schemas/metadata.py:22]: Git-diff cuts off at an unlucky place here:
The schema this method appears in is called [`NoValidationSchema`](https://github.com/tu-graz-library/invenio-records-lom/blob/b3bbf2bbc37e8030a8abdc2b66ed84df6d20158f/invenio_records_lom/services/schemas/metadata.py#L13).
The implications of not validating within something called `NoValidationSchema` should be obvious enough I hope?

P.S. For the record: Without overwriting dump, the following case behaves differently:
```python
# dump `{"a": "0"}`, `NoValidationSchema.dump` should return it as is
Schema().dump({"a": "0"})  # code as previously, uses marshmallow.Schema.dump (from its parent class), returns `{}`
DumpOverwrittenSchema().dump({"a": "0"})  # code as of now, returns `{"a": "0"}`
# i.e. `"a"` isn't included in the dump, despite the configuration `unknown = marshmallow.INCLUDE`
```

P.P.S. should actually be `copy.copy(obj)` for consistency with marshmallow

File: invenio_records_lom/services/schemas/metadata.py [hunk 5]
```diff
@@ -18,50 +18,63 @@ class Meta:
...
         if not isinstance(langstring_inner, dict):
             raise self.make_error("invalid")
         inner_keys = set(langstring_inner)
         if "#text" not in inner_keys:
             raise self.make_error("missing_langstring_key", keys="#text")
-        if "lang" not in inner_keys:
+        if "lang" not in inner_keys and self.lang_exists:
             raise self.make_error("missing_langstring_key", keys="lang")
+        if "lang" in inner_keys and not self.lang_exists:
+            raise self.make_error("")
```
[the remark this answers is not in the document]
Comment (martinobersteiner) [reply invenio_records_lom/services/schemas/metadata.py:64]: Whoops, good catch. Should be the newly added error `"extraneous_lang"`...

File: invenio_records_lom/services/schemas/metadata.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (utnapischtim) [reply invenio_records_lom/services/schemas/metadata.py:22]: yes, it seams obvious, but it would help to add a small note to the `def dump` docstring like. `Overwrite dump to return obj, bypassing validation as class name indicates`. don't put the burden of collecting information to the reader, if you could help them that easily.

File: invenio_records_lom/resources/serializers/schemas.py (hunk 1 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas.py:349]: Noted.
OTOH, `contributors` and `contributes` are pretty similar variable names to have in the same scope...
I wished naming variables was less hard...

File: invenio_records_lom/resources/serializers/schemas.py (hunk 2 above)
Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas.py:315]: For the first line, `metadata.general.title` should actually be a langstring-dict, the original version of defaulting to the empty string `""` made me think it's a plain `str`, which confused me for a bit.

For the second line, falsy values that aren't `None` shouldn't really be able to happen here, so I don't think there's a good reason for using any one over the other here...
`title is None`-comparison is just personal preference here.

File: invenio_records_lom/resources/serializers/schemas.py (hunk 1 above)
Comment (utnapischtim) [reply invenio_records_lom/resources/serializers/schemas.py:349]: naming variables is one of the hardest things for a developer ;)

if you think that there could be a confusion because of similar names then you should name it completely different. but i think that is not necessary in this context.

contributes and contributors is ok for me.

Comment (martinobersteiner) [reply invenio_records_lom/resources/serializers/schemas.py]: I renamed it.

File: invenio_records_lom/services/schemas/metadata.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (martinobersteiner) [reply invenio_records_lom/services/schemas/metadata.py]: Done.
